### PR TITLE
[KEYCLOAK-6008] - Spring Boot does not honour wildcard auth-role

### PR DESCRIPTION
**Pull request by mthirion**
KEYCLOAK-12419
I propose this change to support the wildcard character ('*') in the definition of the SecurityConstraint's authRole property of the "Spring-Boot Keycloak adapter" so that an authentication-only flow can be implemented.  At the moment it's not possible due to the way Undertow manages the wildcard character (SecurityPatchMatches's expandRolesAllowed() ).

Notice that the change was tested against the version 8.0.1 because the version 9.0.0-SNAPSHOT currently returns an invalid Bearer Token error.  The logs showed an Access Token value of "<Base64token>.signature", and I'm not sure whether the error could come from the ".signature" extension added somehow at the end of the token, which is not show in version 8.0.1.

### CODE REVIEW

**Comment by sebastienblanc** [comment]
@mthirion Hi ! Thanks for the PR, does this also fix https://issues.redhat.com/browse/KEYCLOAK-6008 ?

**Comment by mthirion** [comment]
Hi,
Yes it looks the same.  I don't know why I didn't see this Jira. 
That's why I had created the KEYCLOAK-12419 one.

File: adapters/oidc/spring-boot-adapter-core/src/main/java/org/keycloak/adapters/springboot/KeycloakBaseSpringBootConfiguration.java [hunk 1]
```diff
@@ -32,22 +34,26 @@
...
 
+
 /**
  * Keycloak authentication base integration for Spring Boot - base to be extended for particular boot versions.
  */
 public class KeycloakBaseSpringBootConfiguration {
 
-    protected KeycloakSpringBootProperties keycloakProperties;
+	
+	protected KeycloakSpringBootProperties keycloakProperties;
```
Comment (pedroigor) [inline adapters/oidc/spring-boot-adapter-core/src/main/java/org/keycloak/adapters/springboot/KeycloakBaseSpringBootConfiguration.java]: Could you keep indentation and avoid unnecessary changes to the file ?

File: adapters/oidc/spring-boot-adapter-core/src/main/java/org/keycloak/adapters/springboot/KeycloakBaseSpringBootConfiguration.java [hunk 2]
```diff
@@ -76,11 +82,34 @@ public void customize(DeploymentInfo deploymentInfo) {
...
+            		}
+            		deploymentInfo.addSecurityConstraint(allRolesAllowed);
+            	} else // left empty will fall back on default EmptyRoleSemantic.DENY
+            		deploymentInfo.addSecurityConstraint(securityConstraint);
+            	
+            }
             deploymentInfo.addServletExtension(new KeycloakServletExtension());
         }
-
+
```
Comment (pedroigor) [inline adapters/oidc/spring-boot-adapter-core/src/main/java/org/keycloak/adapters/springboot/KeycloakBaseSpringBootConfiguration.java]: Could you keep avoid unnecessary changes to the file?

File: adapters/oidc/spring-boot-adapter-core/src/main/java/org/keycloak/adapters/springboot/KeycloakBaseSpringBootConfiguration.java [hunk 3]
```diff
@@ -105,7 +134,7 @@ public void customize(DeploymentInfo deploymentInfo) {
             return undertowSecurityConstraints;
         }
     }
-
+
```
Comment (pedroigor) [inline adapters/oidc/spring-boot-adapter-core/src/main/java/org/keycloak/adapters/springboot/KeycloakBaseSpringBootConfiguration.java]: Could you keep avoid unnecessary changes to the file?

**Review by pedroigor — CHANGES_REQUESTED** [review]
Just a few comments. LGTM.

**Comment by pedroigor** [comment]
@mthirion In addition to those comments, please change:

* Commit message to: `[KEYCLOAK-6008] - Spring Boot does not honour wildcard auth-role`

File: adapters/oidc/spring-boot-adapter-core/src/main/java/org/keycloak/adapters/springboot/KeycloakBaseSpringBootConfiguration.java (hunk 2 above)
Comment (mthirion) [reply adapters/oidc/spring-boot-adapter-core/src/main/java/org/keycloak/adapters/springboot/KeycloakBaseSpringBootConfiguration.java]: Done. Should be better now.

File: adapters/oidc/spring-boot-adapter-core/src/main/java/org/keycloak/adapters/springboot/KeycloakBaseSpringBootConfiguration.java (hunk 3 above)
Comment (mthirion) [reply adapters/oidc/spring-boot-adapter-core/src/main/java/org/keycloak/adapters/springboot/KeycloakBaseSpringBootConfiguration.java]: Done. Should be better now.

File: adapters/oidc/spring-boot-adapter-core/src/main/java/org/keycloak/adapters/springboot/KeycloakBaseSpringBootConfiguration.java (hunk 1 above)
Comment (mthirion) [reply adapters/oidc/spring-boot-adapter-core/src/main/java/org/keycloak/adapters/springboot/KeycloakBaseSpringBootConfiguration.java]: Done. Should be better now.

**Review by pedroigor — APPROVED** [review]

**Comment by pedroigor** [comment]
@mthirion Thanks.
